src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -356,20 +356,25 @@ public string GetHelpMaml(string helpItemName, string path)
         protected override ProviderInfo Start(ProviderInfo providerInfo)
         {
             // Set the home folder for the user
-            if (providerInfo != null && string.IsNullOrEmpty(providerInfo.Home))
+            if (providerInfo != null)
             {
-                // %USERPROFILE% - indicate where a user's home directory is located in the file system.
-                string homeDirectory = Environment.GetEnvironmentVariable(Platform.CommonEnvVariableNames.Home);
+                providerInfo.PathSeparator = new[] { "\\", "/" };
```

Comment:
Perhaps we already have the array in static cache. Below the same.

Reply:
@iSazonov Good question.  Also, does the FileSystem provider use both separator for *Unix*-like systems as well? Or just the */* character?

I couldn't find anything conclusive on that regard

Reply:
Unix only uses forward slash, I believe. `\` is a valid filename character, isn't it?

Reply:
It seems there is Normalization() method in FileSystem provider that allow to use both in PowerShell

Reply:
@vexx32 I am aware that Unix shells use forward slash only. I guess a better wording for my question would be: Does PowerShell allows the use of both slashes as separators? 😄

Reply:
@renehernandez had a couple folks in the PowerShell Slack/Discord/IRC channels check it out as I don't have a Linux VM on hand at present, and it seems like the standard PS cmdlets do indeed permit backslashes to be used as a path separator even on Linux.

Reply:
@iSazonov regarding the initial question, do you suggest to initialize PathSeparator as a reference of  a static (optionally readonly) array instance?

Reply:
Yes, we need search in our code - perhaps we already have such array.

Reply:
@iSazonov Regarding the name convention, @mklement0  discussed both on the [issue](https://github.com/PowerShell/PowerShell/issues/5544) and he was leaning towards using `PathSeparator`. Perhaps we could get him to expand on this?

Reply:
Yes, we need to discuss that will helps PowerShell Committee.

Reply:
@iSazonov After looking for examples of an array in static cache and not finding anything, I added new static caches in a class next ProviderInfo.

While doing this I realize the need to switch to a `ReadOnlyCollection` instead of a plain array  to avoid situations like the following

```powershell
 $separator = (Get-PSProvider FileSystem).PathSeparator
 $separator[0] = "w"
 (Get-PSProvider FileSystem).PathSeparator | Should -Be @("\", "/") // Fails because it returns @("w", "/")
```

Using `ReadOnlyCollection`, we avoid altogether of the user modifying the PathSeparator property elements

Reply:
@renehernandez you can define a readonly member something like this, to prevent it from being set by anyone outside the provider class:

```csharp
public string[] PathSeparator { get; private set; } = { "/", "\\" }
```

... _if_ I remember my C# semantics correctly 😉 
You may also want to explicitly declare it as a `readonly` member, but I'm not sure how that would work with where this needs to be set. 😄

Reply:
@iSazonov @vexx32 There are two different issues to account for here:

1- `PathSeparator` property setter needs to be internally accessible because the different providers are required to set their corresponding value on the property, which they do on the `Start` method receiving a `ProviderInfo` obj as a parameter.

2- The issue that I highlighted in the previous comment will happens if we use `List` or `array` regardless of the scope of the setter property, because the `ProviderInfo` object gets cached per session and thus, we will be assigning a new reference to the same collection every time that `PathSeparator` property is invoked in the same session and modifications to these references will directly modify the original object (which can be seen in the example code that I posted in the above comment). So we either return a read-only object (such as `ReadOnlyCollection` or we return a new object (`List` or `array`) every time the `PathSeparator` property is invoked.

Reply:
@iSazonov @vexx32 @daxian-dbw Can we resolve this conversation and move forward with the pull request? Or is there anything left that should be changed related to this discussion??

Reply:
@renehernandez Sorry, I lost the conversation.
We could use:
```c#
public IReadOnlyList<string> PathSeparator { get; } = { "/", "\\" };
```
Perhaps @daxian-dbw has wiser advice.

Reply:
@iSazonov @vexx32 @daxian-dbw Another alternative would be to use `ImmutableArray` from [System.Collections.Immutable package](https://www.nuget.org/packages/System.Collections.Immutable/). Thinking now, I think it would be a better approach since it gives you a stronger guarantee of safety in multithreaded scenarios. That may not be a problem for the `PathSeparator` property in reality, but it won't hurt.

Any comments?

Reply:
I think we don't want this if it is an another dll.

Reply:
@iSazonov the [System.Collections.Immutable](https://www.nuget.org/packages/System.Collections.Immutable/) package is already in use on the powershell code in the [AddType.cs](https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs) file, so we wouldn't be introducing a new dependency

Reply:
@SteveL-MSFT @daxian-dbw What is best type for public API here?

Reply:
@SteveL-MSFT @daxian-dbw In my perspective deciding on the best API type is currently the only blocker to resolve this conversation and move forward with this pull request. Any input on it?

Reply:
@SteveL-MSFT @daxian-dbw Any resolution on this topic? I believe this should be relatively simple to tackle